docs/src/modules/java/partials/topic-eventing.adoc

```diff
@@ -27,19 +27,35 @@ The Protobuf rpc method receiving these JSON messages must be set up to receive
 
 [source,protobuf,indent=0]
 ----
-syntax = "proto3";
+include::java:example$java-eventing-shopping-cart/src/main/proto/cart/shopping_cart_topic.proto[tag=receiving-json-messages]
+----
 
-import "google/protobuf/empty.proto";
-import "google/protobuf/any.proto";
-import "akkaserverless/annotations.proto";
 
-package shopping.cart.actions;
+The Java implementation of the method will follow the protobuf definintion and will receive `Any`. In the method implementation. You can then decode the message to the expected typed (eg: `TopicMessage`) using `JsonSupport.decodeJson`.
 
-service ShoppingCartTopicService {
-    rpc JsonFromTopic(google.protobuf.Any) returns (google.protobuf.Empty) {
-        option (akkaserverless.method).eventing.in = {
-            topic:  "shopping-cart-json"
-        };
-    }
-}
+[source,java,indent=0]
+----
+include::java:example$java-eventing-shopping-cart/src/main/java/shopping/cart/api/ShoppingCartTopicServiceAction.java[tag=receiving-json-messages]
+----
+
+The parameter class needs to be annotated as `@Jsonable` and prepared for Jackson JSON parsing by `@BeanProperty` annotations.
+
+[source,java,indent=0]
 ----
+import com.akkaserverless.javasdk.Jsonable;
```

Comment:
Jsonable doesn't exist. Note that I documented the JSON support in https://github.com/lightbend/akkaserverless-java-sdk/pull/323

Reply:
Correct. I will fix it.

Reply:
@patriknw, fixed now. Made it similar to `JsonKeyValueMessage`, but not with snippet include. 
And added a link to the serialization page. 

On next iteration we can use includes and improve this.